src/main/frontend/format/block.cljs

```diff
@@ -38,25 +41,16 @@ and handles unexpected failure."
 (defn parse-block
   ([block]
    (parse-block block nil))
-  ([{:block/keys [uuid content page format] :as block} {:keys [with-id?]
+  ([{:block/keys [uuid content format] :as block} {:keys [with-id?]
                                                         :or {with-id? true}}]
    (when-not (string/blank? content)
      (let [block (dissoc block :block/pre-block?)
            ast (format/to-edn content format nil)
-           blocks (extract-blocks ast content with-id? format)
+           blocks (extract-blocks ast content format {:with-id? with-id?
+                                                      :extract-macros? false})
```

Comment:
When I re-index a graph, I see that `block/extract-blocks` initially run with this option to set true. When we're editing a block, we don't re-calculate here but instead over in `outliner.datascript/transact!`. Why is that and are there any edge cases where macros blocks could get stale? A comment here or in the outliner fn would be helpful

Reply:
The previous problem is that logseq doesn't handle multiple blocks when editing a block yet, so if a block has some macros, it'll be parsed to multiple blocks. To fix that, a lot of functions need to be changed.

`:block/macros` was added so that there's no need to separate the usual page blocks and macro blocks.